### ISSUE TITLE
Print connected cluster.

### ISSUE THREAD

**Issue by guettli** [guettli/check-conditions#12]
**Current state:** 💤 waiting for agent

[Open in AgentLoop UI](https://agentloop.thomas-guettler.de/issues/check-conditions/12)

I have enough to write the plan.

## Summary

Print the connected cluster (server URL + kube-context), the current time, and the kubeconfig source at the start of every run. The client config is loaded once in `RunAllOnce` (`pkg/checkconditions/checkconditions.go:214`) via `clientcmd.NewDefaultClientConfigLoadingRules()`, which already exposes everything needed.

## Implementation

1. **Extract the loading rules so the source is inspectable** — In `RunAllOnce` (`checkconditions.go:214-231`), keep the existing `loadingRules` / `kubeconfig` (`DeferredLoadingClientConfig`) locals but also read:
   - Current context + cluster/server: call `kubeconfig.RawConfig()` (returns `api.Config`). Use `rawConfig.CurrentContext` and `rawConfig.Clusters[rawConfig.Contexts[current].Cluster].Server` for the server URL. `config.Host` is a simpler fallback for the server URL.
   - Kubeconfig source: from `loadingRules` — `loadingRules.ExplicitPath` if set (e.g. `--kubeconfig`, not currently wired), otherwise `loadingRules.GetLoadingPrecedence()` (the file list, driven by `$KUBECONFIG` or the `~/.kube/config` default). Note whether `$KUBECONFIG` env var was set.

2. **Add a small helper** e.g. `printConnectionInfo(kubeconfig clientcmd.ClientConfig, loadingRules *clientcmd.ClientConfigLoadingRules, config *rest.Config)` in `checkconditions.go` that prints one concise block:
   ```
   Cluster:    https://1.2.3.4:6443 (context "kind-kind")
   Kubeconfig: /home/user/.kube/config  (from $KUBECONFIG)
   Time:       2026-08-28 15:04:05 -0700 MST
   ```
   Guard against errors from `RawConfig()` and missing map keys (fall back to `config.Host` / "unknown").

3. **Call it once per invocation, not per retry.** Currently:
   - `all` → `RunAllOnce` (no time/cluster printed today).
   - `while` → `runWhileInner` → `RunAllOnce` in a loop; time is printed per iteration at `checkconditions.go:274-278`.
   - `forever` → `RunForever` loop; time printed at `checkconditions.go:240`.

   To "always print" without spamming on every loop iteration, print the connection block **once at startup**. Cleanest spot: add the print inside `RunAllOnce` but gated by a `sync.Once` (or a bool on `Arguments`) so repeated calls from `while`/`forever` only print it the first time. The existing per-iteration timestamps in the loops stay as-is.

4. **Verbose detail (optional):** under `--verbose`, also print the full `GetLoadingPrecedence()` list so users see every file that was merged, not just the primary.

5. Run `go build ./...`, `go vet ./...`, and `golangci-lint` per `.golangci.yaml`; add/extend a test in `pkg/checkconditions/checkconditions_test.go` for the source-string/formatting helper if it is pure enough to unit-test.

<details>
<summary>📋 Planning questions (from the plan stage)</summary>

## Questions

1. What does "connected cluster" mean for the output — which fields?
   - Option A: server URL only (e.g. `https://1.2.3.4:6443`)
   - Option B: kube-context name only (e.g. `kind-kind`)
   - Option C: both server URL and context name (recommended)
   - Option D: something else — tell me what you have in mind

2. Where/how often should this block be printed?
   - Option A: once at program startup only (first `RunAllOnce`), not repeated in the `while`/`forever` loops (recommended — avoids clutter, keeps existing per-iteration timestamps)
   - Option B: on every loop iteration alongside the existing timestamp
   - Option C: something else

3. For the kubeconfig source, how much detail?
   - Option A: just the primary/effective file path, plus a note when `$KUBECONFIG` is set
   - Option B: the primary path always, and the full merge precedence list only under `--verbose` (recommended)
   - Option C: always print the full precedence list

4. Is a `--kubeconfig` / `--context` flag in scope here? Today the tool relies purely on defaults + `$KUBECONFIG`; the issue only asks to *print* the source, so I'm assuming **no new flags** unless you want them.

</details>

Always print the connected cluster and the current time.

Is it possible to print where the kubeconfig is coming from?

Would be nice.

<details>
<summary>⏱ Timeline (21 steps)</summary>

| When | Event | Duration | Waiting on |
|------|-------|----------|------------|
| … | +1 earlier event | | |
| 08-27 20:28 | label `loop/triage-in-process` set | 1s | 💤 waiting for agent |
| 08-27 20:28 | agent run started | 12m48s | ❌ agent failed |
| 08-27 20:41 | agent run failed | 1s | 💤 waiting for agent |
| 08-27 20:41 | agent run started | 56s | ▶️ agent working |
| 08-27 20:42 | label `NeedSupervisor` set | 4m18s | 💤 waiting for human |
| 08-27 20:46 | label `NeedSupervisor` cleared | 25h58m | 💤 waiting for agent |
| 08-28 22:44 | label `loop/triage` set | 17s | 💤 waiting for agent |
| 08-28 22:44 | label `loop/triage-in-process` set | 1s | 💤 waiting for agent |
| 08-28 22:44 | agent run started | 3m43s | ✅ agent ran |
| 08-28 22:48 | agent run succeeded | 0s | 💤 waiting for agent |
| 08-28 22:48 | label `loop/plan` set | 6s | 💤 waiting for agent |
| 08-28 22:48 | label `loop/plan-in-process` set | 1s | 💤 waiting for agent |
| 08-28 22:48 | agent run started | 4m35s | ✅ agent ran |
| 08-28 22:53 | agent run succeeded | 1s | 💤 waiting for agent |
| 08-28 22:53 | label `loop/plan-questions` set | 42s | 💤 waiting for agent |
| 08-28 22:54 | label `loop/code` set | 5s | 💤 waiting for agent |
| 08-28 22:54 | label `loop/code-in-process` set | 1s | 💤 waiting for agent |
| 08-28 22:54 | agent run started | 12m50s | ✅ agent ran |
| 08-28 23:06 | agent run succeeded | 1s | 💤 waiting for agent |
| 08-28 23:06 | label `loop/code-ci-pending` set | ongoing | 💤 waiting for agent |

</details>

**Comment by guettlibot** [comment]
agentloop: previous agent run did not complete; resuming acpx session (restart 1/3).

Reason:
agentloop: could not read the agent output: open /data/agentloop/check-conditions/issues/12.meta/run-20260827T202820.327747192Z/output.json: no such file or directory

- [old run](https://agentloop.thomas-guettler.de/issues/check-conditions/12/runs/run-20260827T202820.327747192Z)
- [new run](https://agentloop.thomas-guettler.de/issues/check-conditions/12/runs/run-20260827T204109.511721130Z)

**Comment by guettlibot** [comment]
I've verified the current state of the code against the request.

**Decision: queue for planning.**

The requested feature is not implemented. In `pkg/checkconditions/checkconditions.go:214-231` (`RunAllOnce`), the kubeconfig is loaded via `clientcmd.NewNonInteractiveDeferredLoadingClientConfig`, but the connected cluster (current context / server host), the kubeconfig source path, and the current time are never printed for a run. The only summary output is the "Checked N conditions..." line at `checkconditions.go:355`, which contains none of this. The current time is printed only inside the `RunForever`/while-regex loops (`checkconditions.go:240`, `:277`), not for a single check or as connected-cluster info.

The request is small and well-specified: the clientcmd APIs already in use expose everything needed (`loadingRules.GetLoadingPrecedence()` for the kubeconfig source, `kubeconfig.RawConfig().CurrentContext` for the cluster/context, `config.Host` for the server), so a coding agent can plan this directly. No open issue duplicates it (checked #4–#34 via `gh issue list`).

<sub>agentloop · agent: claude</sub>

**Comment by guettlibot** [comment]
Answers:

- **Q1 = C** — print **both** the server URL (e.g. `https://1.2.3.4:6443`) and the kube-context name (e.g. `kind-kind`). Together they unambiguously identify what you're about to check.
- **Q2 = A** — print it **once at startup** (first `RunAllOnce`), not on every loop iteration. The cluster/context doesn't change between iterations, and the existing per-iteration timestamp already covers "current time". This satisfies the "always print the connected cluster" ask without cluttering the forever-loop output.
- **Q3 = B** — always print the primary/effective kubeconfig path (with a note when `$KUBECONFIG` is set); show the full merge precedence list only under `--verbose`.
- **Q4 = no new flags** — correct assumption. The issue only asks to *print* the source, so don't add `--kubeconfig`/`--context` here; that can be a separate issue if ever wanted.

Proceed under C / A / B / no-new-flags.

**Comment by guettlibot** [comment]
PR opened: https://github.com/guettli/check-conditions/pull/39

## Summary

Implemented the request from issue #12 — the tool now always prints the connected cluster and current time at startup, and shows where the kubeconfig comes from.

**What it prints** (once per run, before the checks):
```
Cluster:    https://127.0.0.1:6443 (context "kind-kind")
Kubeconfig: /home/user/.kube/config
Time:       2026-08-28 15:04:05 +0000 UTC
```

**Details:**
- **Cluster** — server URL taken from the current context's cluster (falling back to `config.Host`) plus the kube-context name.
- **Kubeconfig source** — yes, this is possible: I read it from `clientcmd`'s loading rules. It shows the explicit path if one is set, otherwise the loading-precedence file list, and appends `(from $KUBECONFIG)` when that env var is in play.
- **Time** — same format already used in the loops.

The block is gated by a new `connectionInfoPrinted` flag on `Arguments` so the `while`/`forever` loops (which call `RunAllOnce` repeatedly) print it only once rather than every iteration. All three subcommands (`all`, `while`, `forever`) route through `RunAllOnce`, so they all get it.

**Verified:** `go build`, `go vet`, `go test ./...` (added `TestKubeconfigSource` covering explicit path, precedence list, and the `$KUBECONFIG` note), and `golangci-lint run` — 0 issues.

<sub>agentloop · agent: claude</sub>

**Comment by guettlibot** [comment]
agentloop: closing this issue — its pull request (https://github.com/guettli/check-conditions/pull/39) has been merged.
